File: OfflineFetcher/app/src/main/java/de/jbamberger/offlineclient/ui/components/ListItem.java
package de.jbamberger.offlineclient.ui.components;

import android.support.annotation.LayoutRes;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

public interface ListItem {
    Object getListener();

    Object getObject();

    @LayoutRes
    int getLayoutRes();
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/ui/components/ListItem.kt
package de.jbamberger.offlineclient.ui.components

import android.support.annotation.LayoutRes

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

interface ListItem {
    val listener: Any?

    val obj: Any?

    @get:LayoutRes
    val layoutRes: Int
}